### ISSUE TITLE
AttributeError: 'Sequence' object has no attribute '__bytes__' / '__unicode__'

### ISSUE THREAD

**Issue by Poddster** [wbond/asn1crypto#26]
`print()` used on a `Sequence` will raise an exception.

Is this a code bug or a documentation issue?

 e.g. are subclasses of `Sequence` meant to implement `__bytes__` and `__unicode__` ? 

I notice that subclasses of `Concat` can use its implementation -- though I note that the py2 and py3 will give inconsistent printouts, so I'm not sure if you intended it to be this way or not.

Thanks.

test.py:
```python
#!/usr/bin/env python2
# coding=utf-8
from __future__ import absolute_import, division, print_function
import asn1crypto.core

class MySequence(asn1crypto.core.Sequence):
    _fields = [
        ('abc', asn1crypto.core.PrintableString),
    ]

keyinfo_dict= {'abc': u"1234"}
ki = MySequence(keyinfo_dict)
print(repr(ki.dump()))
print(ki.debug())
print(ki)
```

output:
```text
$ python 2 ./test.py 
'0\x06\x13\x041234'
  __main__.MySequence Object #140130113838608
    Header: 0x3006
      constructed universal tag 16
    Data: 0x130431323334
      Field "abc"
        asn1crypto.core.PrintableString Object #140130113838736
          Header: 0x1304
            primitive universal tag 19
          Data: 0x31323334
            Native: 1234
None
Traceback (most recent call last):
  File "./test.py", line 15, in <module>
    print(ki)
  File "/usr/local/lib/python2.7/dist-packages/asn1crypto/core.py", line 290, in __str__
    return self.__bytes__()
AttributeError: 'MySequence' object has no attribute '__bytes__'

$ python3 ./test.py 
b'0\x06\x13\x041234'
  __main__.MySequence Object #140594304391152
    Header: 0x3006
      constructed universal tag 16
    Data: 0x130431323334
      Field "abc"
        asn1crypto.core.PrintableString Object #140594304492824
          Header: 0x1304
            primitive universal tag 19
          Data: 0x31323334
            Native: 1234
None
Traceback (most recent call last):
  File "./test.py", line 15, in <module>
    print(ki)
  File "/usr/local/lib/python3.5/dist-packages/asn1crypto/core.py", line 292, in __str__
    return self.__unicode__()
AttributeError: 'MySequence' object has no attribute '__unicode__'
```

**Comment by wbond** [comment]
I don’t think it is really either. I don’t know what I would expect to happen if someone “prints” a value, and thus I don’t really guess.

`.dump()` returns the DER-encoded value. `.native` returns a native Python representation. `.debug()` prints out a bunch of details about the encoding.

Which of the three is the correct value to assume?

**Comment by Poddster** [comment]
>Which of the three is the correct value to assume?

No idea, I just don't want a generic bit of doing using `str()` to except :)

In general, `__repr__` is unambiguous, `__str__` is human readable. So I would suggest use `.native` for `__repr__` and `.debug` for `__str__`?

**Comment by wbond** [comment]
The output from `.debug()` is pretty lengthy, and mostly useful for investigating why encoding or parsing isn't working as expected.

I think after mulling this over I'm inclined to have all non-AbstractString types use `__repr__()` output for `__unicode__()`.

**Comment by wbond** [comment]
This should be resolved by 13c875d634ce5994b3e188e889846ba9ee8e6f39. All `core` classes now have a default `__bytes__()` and `__unicode__()` that output the equivalent of `repr()`.
